pkg/adapterManager/manager2.go

```diff
@@ -0,0 +1,439 @@
+						inst.Params.(proto.Message),
+						requestBag, m.mapper,
+						call.handlerState.Handler)
+					return &result2{err, &resp, call}
+				},
+			})
+		}
+	}
+
+	cres, err := m.run(ctx, ra)
+	var res *adapter.CheckResult
+	if cres != nil {
+		res = cres.(*adapter.CheckResult)
+	}
+	return res, err
+}
+
+// Quota dispatches to the set of aspects associated with the Quota API method
+// Config validation ensures that things are consistent.
+// If they are not, we should continue as far as possible on the runtime path
+// before aborting. Returns an error if any of the adapters return an error.
+// If not the results are combined to a single CheckResult.
+// Dispatcher#Quota.
+func (m *Manager2) Quota(ctx context.Context, requestBag attribute.Bag,
+	qma *aspect.QuotaMethodArgs) (*adapter.QuotaResult2, error) {
+	rt := m.RuntimeState()
+	// cannot be null by construction
+
+	calls, err := fetchDispatchInfo(requestBag, adptTmpl.TEMPLATE_VARIETY_QUOTA, rt, m.templateRepo,
+		func(_ *HandlerState, inst *cpb.Instance) bool {
+			return inst.Name == qma.Quota
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	ra := make([]*runArg, 0, 10)
+
+	// TODO reduce context deadline before sending it down to adapters.
+	for _, call := range calls {
+		for _, inst := range call.instances {
+			ra = append(ra, &runArg{call, func(ctx context.Context) *result2 {
+				resp, err := call.processor.ProcessQuota(ctx, inst.Name,
+					inst.Params.(proto.Message), requestBag, m.mapper, call.handlerState.Handler,
+					adapter.QuotaRequestArgs{
+						DeduplicationID: qma.DeduplicationID,
+						QuotaAmount:     qma.Amount,
+						BestEffort:      qma.BestEffort,
+					})
+				return &result2{err, &resp, call}
+			}})
+		}
+	}
+
+	qres, err := m.run(ctx, ra)
+	var res *adapter.QuotaResult2
+	if qres != nil {
+		res = qres.(*adapter.QuotaResult2)
+	}
+	return res, err
+}
+
+// Preprocess runs the first phase of adapter processing before any other adapters are run.
+// Attribute producing adapters are run in this phase.
+func (m *Manager2) Preprocess(ctx context.Context, requestBag attribute.Bag, responseBag *attribute.MutableBag) error {
+	// FIXME
+	return errors.New("not implemented")
+}
+
+// combineResults2 combines results
+func combineResults2(results []*result2) (adapter.Result, error) {
+	var res adapter.Result
+	var err *multierror.Error
+	var buf *bytes.Buffer
+	code := rpc.OK
+
+	for _, rs := range results {
+		if rs.err != nil {
+			err = multierror.Append(err, rs.err)
+		}
+
+		if rs.res == nil { // When there is no return value like ProcessReport().
+			continue
+		}
+		if res == nil {
+			res = rs.res
+		} else {
+			res.Combine(rs.res)
+		}
+		st := rs.res.GetStatus()
+		if !status.IsOK(st) {
+			if buf == nil {
+				buf = pool.GetBuffer()
+				// the first failure result's code becomes the result code for the output
+				code = rpc.Code(st.Code)
+			} else {
+				buf.WriteString(", ")
+			}
+
+			buf.WriteString(rs.callinfo.handlerName + ":" + st.Message)
```

Comment:
Sure, I am fine with that. Just to clarify I was referring to adding the code to the string rs.callinfo.handlerName + ":" + st.Code + " :" + st.Message + "\n"